### PR TITLE
fix(firebase_messaging): ensure plugin callback registered

### PR DESCRIPTION
**Pull request by Eddydpyl**
The fact that the isolate is already running should not prevent the callback from being registered.

## Description

The onBackgroundMessage callback stops triggering after app restart (see related issues). This seems to be caused by the pluginRegistrantCallback not being registered every time the FcmDartService#start method is called, due to the fact that the isolate is already running. Removing the check for the isolate not running solves the problem, and I'm yet to observe any side effects. That being said, I cannot guarantee this is the best way to handle the issue.

## Related Issues

Solves the issue descrived in https://github.com/FirebaseExtended/flutterfire/issues/1709 and might also solve https://github.com/FirebaseExtended/flutterfire/issues/1590.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by hb0nes** [comment]
Very nice, merge this ASAP.
We need (and use) this change.

**Comment by Finn0811** [comment]
I'd also like to see it merged 👍

**Comment by brokenalarms** [comment]
bump for this please, just spent the past few hours trying to get this working, background messages was the first thing I tried to get working after installing this plugin and it doesn't work without this change.

**Comment by Eddydpyl** [comment]
@brokenalarms I've updated CHANGELOG.md and it looks like that was necessary for the code owners to be notified, my bad. In the mean time, you could use my fork (I've updated it just now): https://github.com/Eddydpyl/flutterfire

**Comment by brokenalarms** [comment]
Awesome, thanks!

[Daniel - Chat @ Spike](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=gc8da)	[gc8da]

On April 6, 2020 at 6:51 GMT, Eduardo del Palacio Lirola <notifications@github.com> wrote:

[@brokenalarms](https://github.com/brokenalarms) I've updated CHANGELOG.md and it looks like that was necessary for the code owners to be notified, my bad. In the mean time, you could use my fork (I've updated it just now): https://github.com/Eddydpyl/flutterfire

**Comment by batuyazilim** [comment]
so this is the solition ? i have same problem 
how can i solve this problem?

**Comment by windows7lake** [comment]
why not merge?

**Comment by loolooii** [comment]
Why is this PR not being reviewed?

File: packages/firebase_messaging/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 6.0.15
+
+* Register pluginRegistrantCallback every FcmDartService#start.
```
Comment (Salakar) [inline packages/firebase_messaging/CHANGELOG.md:3]:
```suggestion
* Fix - register `pluginRegistrantCallback` on every `FcmDartService#start` call.
```

**Comment by Salakar** [comment]
@googlebot I consent.

**Comment by Salakar** [comment]
Thanks for sending this PR. The change makes sense to me and as far as I can tell this is the correct fix.

I've gone ahead and published `firebase_messaging` version `6.0.15`. Will now merge (ignoring failing `publishable` CI step - have a separate PR in progress to fix this for all plugins).

**Comment by Fbrusca** [comment]
@Eddydpyl  In my case "onBackgroundMessage" still works inconsistently. Sometimes it works correctly and sometimes it returns "Service took too long to process intent: com.google.android.c2dm.intent.RECEIVE App may get closed."

Could someone be kind enough to pass me the plugin registry settings for embedding v2?

```

class Application: FlutterApplication(), PluginRegistrantCallback {
    override fun onCreate() {
        super.onCreate()
        FlutterFirebaseMessagingService.setPluginRegistrant(this);
    }

    override fun registerWith(registry: PluginRegistry) {
        CustomPluginRegistrant.registerWith(registry)
    }
}

class CustomPluginRegistrant {
    companion object {
        fun registerWith(registry: PluginRegistry) {
            if (alreadyRegisteredWith(registry)) {
                return
            }
            FirebaseMessagingPlugin.registerWith(registry.registrarFor("io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin"))
            FlutterLocalNotificationsPlugin.registerWith(registry.registrarFor("com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin"));
        }

        fun alreadyRegisteredWith(registry: PluginRegistry): Boolean {
            val key: String = FirebaseCloudMessagingPluginRegistrant::class.java.getCanonicalName()
            if (registry.hasPlugin(key)) {
                return true
            }
            registry.registrarFor(key)
            return false
        }
    }
}

```

**Comment by Eddydpyl** [comment]
@Fbrusca I'm afraid I don't know how to help you with this. You should probably open a new issue.

**Pull request merged**